src/controllers/controlpickermenu.cpp

```diff
@@ -813,52 +791,43 @@ ControlPickerMenu::ControlPickerMenu(QWidget* pParent)
+void ControlPickerMenu::addSingleControl(QString group, QString control,
+                                         QString title, QString description,
+                                         QMenu* pMenu, QString prefix,
+                                         QString actionTitle) {
+    int controlIndex = m_controlsAvailable.size();
 
-    if (addReset) {
-        QString resetDescription = QString("%1 (%2)").arg(title,
-                                                          m_resetStr);
-        QString resetControl = QString("%1_set_default").arg(control);
-        QAction* pResetAction = pMenu->addAction(resetDescription,
-                                                 &m_actionMapper, SLOT(map()));
-        m_actionMapper.setMapping(pResetAction, m_controlsAvailable.size());
-        addAvailableControl(ConfigKey(group, resetControl),
-                            resetDescription, resetDescription);
+    if (actionTitle.isEmpty()) {
+        actionTitle = title;
     }
-}
 
-void ControlPickerMenu::addPrefixedControl(QString group, QString control,
-                                           QString controlTitle,
-                                           QString controlDescription, QString descriptionPrefix,
-                                           QMenu* pMenu, bool addReset) {
-    QAction* pAction = pMenu->addAction(controlTitle, &m_actionMapper, SLOT(map()));
-    m_actionMapper.setMapping(pAction, m_controlsAvailable.size());
+    auto pAction = new QAction(actionTitle, pMenu);
+    connect(pAction, &QAction::triggered,
+            this, [this, controlIndex] { controlChosen(controlIndex); });
+    pMenu->addAction(pAction);
 
-    QString title = QString("%1: %2").arg(descriptionPrefix, controlTitle);
-    QString description = QString("%1: %2").arg(descriptionPrefix,
-                                                controlDescription);
-    addAvailableControl(ConfigKey(group, control), title, description);
+    if (prefix.isEmpty()) {
+        addAvailableControl(ConfigKey(group, control), title, description);
+    } else {
+        QString prefixedTitle = QString("%1: %2").arg(prefix, title);
+        QString prefixedDescription = QString("%1: %2").arg(prefix, description);
+        addAvailableControl(ConfigKey(group, control), prefixedTitle, prefixedDescription);
+    }
+}
+
+void ControlPickerMenu::addControl(QString group, QString control, QString title,
+                                   QString description,
+                                   QMenu* pMenu,
+                                   bool addReset,
+                                   QString prefix) {
+    addSingleControl(group, control, title, description, pMenu, prefix);
 
     if (addReset) {
-        QString resetMenuTitle = QString("%1 (%2)").arg(controlTitle, m_resetStr);
-        QString resetMenuDescription = QString("%1 (%2)").arg(controlDescription, m_resetStr);
+        QString resetTitle = QString("%1 (%2)").arg(title, m_resetStr);
```

Comment:
Is this giving more performance? Because I think it's less readable.

Reply:
Yes, because it does not create the intermediate "%1 (%2)" string from char[] and there is no string parsing for %. But readability is also a high value ... I don't mind.